### PR TITLE
Use platform-agnostic library names in Conan recipe

### PR DESCRIPTION
**Pull request by thejohnfreeman**
Dependents like validator-keys-tool are not correctly linking to libxrpl on Windows because the library filenames are different. I believe that using the platform-agnostic names (as the linker knows them) will work. I kindly ask that @ximinez please test for me with these steps:

1. Clone this branch.
2. `conan export .`
3. Try to build validator-keys-tool against dependency xrpl/2.0.0-b4 (which should be the version that was exported in step 2).

### CODE REVIEW

**Review by ximinez — CHANGES_REQUESTED** [review]
With the change to the library name below, and two changes to your branch in validator-keys-tool (https://github.com/ripple/validator-keys-tool/pull/44), everything seems to work.

Changes:
1. Obviously, change the dependency in `conanfile.txt` to match the rippled version in this branch (`2.0.0-b4`).
2. Remove the `beast::uint_test::dstream` debugger output class. I don't know when it stopped working, so it's easier to just take it out for now.

Those changes are in commit https://github.com/ximinez/validator-keys-tool/commit/7a70e66892c5e0fafab6c8d68c4b1130c4d4efec

File: conanfile.py [hunk 1]
```diff
@@ -147,9 +147,9 @@ def package(self):
     def package_info(self):
         libxrpl = self.cpp_info.components['libxrpl']
         libxrpl.libs = [
-            'libxrpl_core.a',
-            'libed25519.a',
-            'libsecp256k1.a',
+            'xrpl_core',
+            'ed25519',
+            'secp256k1',
```
[the remark this answers is not in the document]
Comment (thejohnfreeman) [reply conanfile.py:152]: Did you test that on Linux?

[the remark this answers is not in the document]
Comment (ximinez) [reply conanfile.py:152]: I'll do that now

[the remark this answers is not in the document]
Comment (ximinez) [reply conanfile.py:152]: Well, that's disappointing
```
-- Using Conan toolchain: /home/ed/dev/validator-keys-tool/build/conan/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- NIH-EP cache path: /home/ed/dev/nih_c/unix_makefiles/Clang_14.0.0/Debug
-- Conan: Component target declared 'xrpl::libxrpl'
-- Conan: Target declared 'xrpl::xrpl'
CMake Error at build/conan/cmakedeps_macros.cmake:39 (message):
  Library 'libsecp256k1' not found in package.  If 'libsecp256k1' is a system
  library, declare it with 'cpp_info.system_libs' property
Call Stack (most recent call first):
  build/conan/xrpl-Target-debug.cmake:24 (conan_package_library_targets)
  build/conan/xrplTargets.cmake:26 (include)
  build/conan/xrpl-config.cmake:10 (include)
  CMakeLists.txt:22 (find_package)
```

[the remark this answers is not in the document]
Comment (ximinez) [reply conanfile.py:152]: And then in Visual Studio _without_ the change:
```
-- Using Conan toolchain: C:/Dev/validator-keys-tool/build/conan.msvc19/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19045.
-- NIH-EP cache path: C:/dev/nih_c/vs_16_2019/MSVC_19.29.30152.0
-- Conan: Component target declared 'xrpl::libxrpl'
-- Conan: Target declared 'xrpl::xrpl'
CMake Error at build/conan.msvc19/cmakedeps_macros.cmake:39 (message):
  Library 'secp256k1' not found in package.  If 'secp256k1' is a system
  library, declare it with 'cpp_info.system_libs' property
Call Stack (most recent call first):
  build/conan.msvc19/xrpl-Target-debug.cmake:24 (conan_package_library_targets)
  build/conan.msvc19/xrplTargets.cmake:26 (include)
  build/conan.msvc19/xrpl-config.cmake:16 (include)
  CMakeLists.txt:22 (find_package)
```
Basically the same error.

**Comment by intelliot** [comment]
note: the following are blockers to merging #4831

> With the change to the library name below, and two changes to your branch in validator-keys-tool ([ripple/validator-keys-tool#44](https://github.com/ripple/validator-keys-tool/pull/44)), everything seems to work.
> 
> Changes:
> 
> 1. Obviously, change the dependency in `conanfile.txt` to match the rippled version in this branch (`2.0.0-b4`).
> 2. Remove the `beast::uint_test::dstream` debugger output class. I don't know when it stopped working, so it's easier to just take it out for now.
> 
> Those changes are in commit [ximinez/validator-keys-tool@7a70e66](https://github.com/ximinez/validator-keys-tool/commit/7a70e66892c5e0fafab6c8d68c4b1130c4d4efec)

**Comment by intelliot** [comment]
notes:
- the secp256k1 dependency is named `secp256k1` on linux, but named `libsecp256k1` on Windows
- need to figure out what names are actually being created on each platform
- figure out why they are not `libname.a` on Linux and `name.lib` on Windows
- fix CMake to produce those names and not any other names (with different names for shared libraries)

**Review by ximinez — APPROVED** [review]
With the latest changes, I'm able to build https://github.com/ripple/validator-keys-tool/pull/44 on both Windows and Linux

**Comment by intelliot** [comment]
`Will Need Documentation`: 
1. make it easy for developers to learn how to depend on libxrpl; 
2. prospective validator operators should be able to build validator-keys-tool

**Review by legleux — APPROVED** [review]

**Comment by intelliot** [comment]
note: this should be merged prior to https://github.com/ripple/validator-keys-tool/pull/44 (the version number used in that PR will need to be updated)

@thejohnfreeman at your convenience, please:
1. Bring this branch up-to-date with the base branch (`develop`)
2. Indicate when, from your perspective, this PR is ready to merge. (When it's ready, we'll put the `Passed` label on it.)

**Comment by intelliot** [comment]
Suggested commit message:
```
fix(libxrpl): change library names in Conan recipe (#4831)

Use consistent platform-agnostic library names on all platforms.

Fix an issue that prevents dependents like validator-keys-tool from
linking to libxrpl on Windows.

It is bad practice to change the binary base name depending on the
platform. CMake already manipulates the base name into a final name that
fits the conventions of the platform. Linkers accept base names on the
command line and then look for conventional names on disk.
```

**Pull request merged**